cachi2/core/package_managers/bundler/parser.py

```diff
@@ -0,0 +1,133 @@
+import json
+import logging
+import re
+import subprocess
+from pathlib import Path
+from typing import Annotated, Optional, Union
+
+import pydantic
+
+from cachi2.core.errors import PackageManagerError, PackageRejected, UnexpectedFormat
+from cachi2.core.rooted_path import PathOutsideRoot, RootedPath
+from cachi2.core.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+GEMFILE = "Gemfile"
+GEMFILE_LOCK = "Gemfile.lock"
+
+AcceptedUrl = Annotated[
+    pydantic.HttpUrl,
+    pydantic.UrlConstraints(allowed_schemes=["https"]),
+]
+
+
+class _GemMetadata(pydantic.BaseModel):
+    """
+    Base class for gem metadata.
+
+    Attributes:
+        name:       The name of the gem.
+        version:    The version of the gem.
+    """
+
+    name: str
+    version: str
+
+
+class GemDependency(_GemMetadata):
+    """
+    Represents a gem dependency.
+
+    Attributes:
+        source:     The source URL of the gem.
+        checksum:   The checksum of the gem.
+    """
+
+    source: str
+    checksum: Optional[str] = None
+
+
+class GitDependency(_GemMetadata):
+    """
+    Represents a git dependency.
+
+    Attributes:
+        url:        The URL of the git repository.
+        revision:   Commit hash.
+        branch:     The branch of the git repository.
+    """
+
+    url: AcceptedUrl
+    revision: str
```

Comment:
Obligatory "Not a blocker" marker. 

It has just struck me, why a field validator and not just `revision: str = Field(pattern=r"^[a-fA-F0-9]{40}$")`?